crates/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -204,21 +203,21 @@ where
     /// installed rather than a bool.
     async fn install(
         &mut self,
-        packages: Vec<FloxPackage>,
-        nix: &NixCommandLine,
-        system: System,
-    ) -> Result<bool, EnvironmentError2> {
+        packages: Vec<String>,
+        _nix: &NixCommandLine,
+        _system: System,
+    ) -> Result<InstallationAttempt, EnvironmentError2> {
         let current_manifest_contents = self.manifest_content()?;
-        let new_manifest_contents =
-            flox_nix_content_with_new_packages(&current_manifest_contents, packages)?;
-        match new_manifest_contents {
-            ManifestContent::Unchanged => return Ok(false),
-            ManifestContent::Changed(contents) => {
-                self.transact_with_manifest_contents(contents, nix, system)
-                    .await?;
-                Ok(true)
-            },
+        let installation = insert_packages(&current_manifest_contents, packages.iter().cloned())
+            .map(|insertion| InstallationAttempt {
+                new_manifest: insertion.new_toml.map(|toml| toml.to_string()),
+                already_installed: insertion.already_installed,
+            })?;
+        if installation.new_manifest.is_some() {
+            // TODO: enable transactions once build is re-implemented
+            // self.transact_with_manifest_contents(toml.to_string(), nix, system).await?;
```

Comment:
suggestion: write manifest in `Environment` impl

even without transaction, I think the modifiying part should be implemented here rather than in the command impl.
The point of the trait is so we _dont_ need to know the impl details in the command method.

Reply:
I'd rather handle that at the same time as implementing transactions. This works for now.